include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -0,0 +1,295 @@
+      behaviors_.front() = std::move(b);
+    } else
+    {
+      LOG_WARN(
+          "%s behavior could not be initialized and will not be set as default behavior.",
+          b->get_name().c_str());
+    }
+  }
+
+  /**
+   * @brief Reset the behaviors list back to default
+   */
+  void ResetBehaviors()
+  {
+    if (behaviors_.size() > 1)
+    {
+      behaviors_.erase(behaviors_.begin() + 1, behaviors_.end());
+      names_.erase(names_.begin() + 1, names_.end());
+    }
+  }
+
+  /**
+   * @brief Update the current behavior
+   * @param robot shared pointer to robot behavior is executing on
+   */
+  virtual void Update()
+  {
+    behaviors_[selected_idx]->ThreadSafeProcessInput();
+    behaviors_[selected_idx]->Update();
+    behaviors_[selected_idx]->ProcessOutput();
+    if (switching_behaviors_
+        && behaviors_[selected_idx]->ReadyToSwitch(*behaviors_[next_idx_]))
+    {
+      behaviors_[selected_idx]->set_inactive();  // mark previous behavior as inactive
+      behaviors_[next_idx_]->Begin(*behaviors_[selected_idx]);  // begin new behavior
+      behaviors_[next_idx_]->set_active();  // mark new behavior as active
+      selected_idx = next_idx_;  // update selected behavior to next
+      switching_behaviors_ = false;  // no longer transitioning behaviors
+      std::fprintf(stdout,
+                   "[INFO][BehaviorManager] Switched to behavior %d.\n",
+                   selected_idx);
+    }
+  }
+
+  /**
+   * @brief Prints a list of behaviors currently in `behaviors_`
+   * @param stream output stream
+   */
+  void PrintBehaviorList(FILE *stream = stdout) const
+  {
+    std::fprintf(stream, "+--------------------------------+\n");
+    std::fprintf(stream, "| Control Loop Behaviors         |\n");
+    std::fprintf(stream, "| ----------------------         |\n");
+    for (unsigned int i = 0; i < names_.size(); i++)
+    {
+      std::fprintf(stream, "| %3d. %-25s |\n",
+                   i, names_[i].c_str());
+    }
+    std::fprintf(stream, "+--------------------------------+\n");
+  }
+
+  /**
+   * @brief Accessor for index of selected behavior
+   * @return selected behavior index
+   */
+  [[nodiscard]] unsigned int get_selected_behavior() const
+  {
+    return selected_idx;
+  }
+
+  /**
+   * @brief Accessor for names of behaviors in `behaviors_`
+   * @param behavior_idx index of behavior
+   * @return name of behavior at index `behavior_idx` in `behaviors_`
+   */
+  [[nodiscard]] std::string get_behavior_name(int behavior_idx) const
+  {
+    return behaviors_[behavior_idx]->get_name();
+  }
+
+  /**
+   * @brief Accessor for name of selected behavior
+   * @return selected behavior name
+   */
+  [[nodiscard]] std::string get_sel_behavior_name() const
+  {
+    return get_behavior_name(selected_idx);
+  }
+
+  /**
+   * @brief Accessor for all behavior names in `behaviors_`
+   * @return vector of names of behaviors
+   */
+  [[nodiscard]] std::vector<std::string> get_behavior_list() const
```

Comment:
`get_behavior_names.

Current function name makes me think its the vector of behaviors.

Reply:
Renamed in d5a6010ee947e62b6bbd122659ed3fcc02ff3224